benchmarks/report.py

```diff
@@ -0,0 +1,50 @@
+import csv
+from abc import ABC, abstractclassmethod
+from dataclasses import dataclass, fields
+from statistics import mean
+from typing import Dict, list, tuple
+
+import numpy as np
+
+duration = int
+
+
+@dataclass
```

Comment:
@NivekT what do you think about this kind of reporting instead? Probably still a few bugs around but just wanna make sure I'm not overengineering things